src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.cs

```diff
@@ -390,14 +389,62 @@ private static bool IsValidIssuer(string validIssuerTemplate, string tenantId, s
 
             if (validIssuerTemplate.Contains(TenantIdTemplate))
             {
-                return validIssuerTemplate.Replace(TenantIdTemplate, tenantId) == actualIssuer;
+                return IssuersWithTemplatesAreEqual(validIssuerTemplate, TenantIdTemplate, actualIssuer, tenantId);
```

Comment:
Instead of `Contains`, if we use `IndexOf`, we can find the place where `TenantIdTemplate` exists in the string in the first pass and don't need to search for it again in `IssuersWithTemplatesAreEqual`.

Reply:
Great point, done